### PR TITLE
feat(policy): enable support for 10k+ policies

### PR DESCRIPTION
**Pull request by david-leifker**
Fetching policies was previously limited to 30 policy pages. This PR increases to 1000 by default. Secondly, moving to the scroll API which performs better with deep pagination in the 10k+ policy counts.

The graphql API changes are not included so we're forced to stay compatible with the current start/offset and count by paginating in the background when needed for the UI.

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-service/auth-impl/src/main/java/com/datahub/authorization/PolicyFetcher.java [hunk 1]
```diff
@@ -38,22 +41,53 @@ public class PolicyFetcher {
...
+      } catch (Exception e) {
+        throw new RuntimeException("Failed to list policies", e);
+      }
+    });
   }
 
-  public PolicyFetchResult fetchPolicies(int start, int count, String query, Authentication authentication)
+  public PolicyFetchResult fetchPolicies(int count, @Nullable String scrollId, Authentication authentication)
+          throws RemoteInvocationException, URISyntaxException {
+    return fetchPolicies("", count, scrollId, authentication);
```
Comment (pedro93) [inline metadata-service/auth-impl/src/main/java/com/datahub/authorization/PolicyFetcher.java:76]: Shouldn't this be use a star query?

File: metadata-service/auth-impl/src/main/java/com/datahub/authorization/PolicyFetcher.java [hunk 2]
```diff
@@ -38,22 +41,53 @@ public class PolicyFetcher {
...
       throws RemoteInvocationException, URISyntaxException {
-    log.debug(String.format("Batch fetching policies. start: %s, count: %s ", start, count));
-    // First fetch all policy urns from start - start + count
-    SearchResult result =
-        _entityClient.search(POLICY_ENTITY_NAME, query, null, POLICY_SORT_CRITERION, start, count, authentication,
-            new SearchFlags().setFulltext(true));
+    log.debug(String.format("Batch fetching policies. count: %s, scroll: %s", count, scrollId));
+
+    // First fetch all policy urns
+    ScrollResult result = _entityClient.scrollAcrossEntities(List.of(POLICY_ENTITY_NAME), query, null, scrollId,
```
Comment (pedro93) [inline metadata-service/auth-impl/src/main/java/com/datahub/authorization/PolicyFetcher.java:84]: How long is ES keeping the scrollId alive for? Are we expecting pulling in the 10k+ policies within a single graphQL call?

**Comment by pedro93** [comment]
Please check the failing CI

File: metadata-service/auth-impl/src/main/java/com/datahub/authorization/PolicyFetcher.java (hunk 1 above)
Comment (david-leifker) [reply metadata-service/auth-impl/src/main/java/com/datahub/authorization/PolicyFetcher.java:76]: Both are work the same way

File: metadata-service/auth-impl/src/main/java/com/datahub/authorization/PolicyFetcher.java (hunk 2 above)
Comment (david-leifker) [reply metadata-service/auth-impl/src/main/java/com/datahub/authorization/PolicyFetcher.java:84]: The general search_after scroll id doesn't work like PIT which has a timeout. In this mode, ES uses sorting and resumes from the position based on the sort criteria on each fetch.

**Comment by david-leifker** [comment]
CI failures unrelated to the changes in this PR

**Pull request merged**